modin/numpy/test/test_array_logic.py

```diff
@@ -0,0 +1,118 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+import numpy
+
+import modin.numpy as np
+
+
+small_arr_c_2d = numpy.array(
+    [
+        [1j, 1, 0, -numpy.inf, numpy.inf, 0.5],
+        [1 + 1.1j, numpy.nan, 0, numpy.nan, 2, 0.3],
+    ]
+)
+small_arr_c_1d = numpy.array([numpy.nan, 0, -numpy.inf, numpy.inf, 5, -0.1, 1 + 1.1j])
+
+small_arr_r_2d = numpy.array(
+    [[1, 0, -numpy.inf, numpy.inf, 0.5], [numpy.nan, 0, numpy.nan, 2, 0.3]]
+)
+small_arr_r_1d = numpy.array([numpy.nan, 0, -numpy.inf, numpy.inf, 5, -0.1])
+
+
+@pytest.mark.parametrize("operand_shape", [100, (3, 100)])
+@pytest.mark.parametrize("operator", ["any", "all"])
+@pytest.mark.parametrize("axis", [None, 0, 1], ids=["axis=None", "axis=0", "axis=1"])
+def test_unary_with_axis(operand_shape, operator, axis):
+    if isinstance(operand_shape, int) and axis == 1:
+        pytest.skip("cannot use axis=1 on 1D arrays")
+    x1 = numpy.random.randint(-100, 100, size=operand_shape)
+    numpy_result = getattr(numpy, operator)(x1, axis=axis)
+    x1 = np.array(x1)
+    modin_result = getattr(np, operator)(x1, axis=axis)
+    # Result may be a scalar
+    if isinstance(modin_result, np.array):
+        modin_result = modin_result._to_numpy()
+    numpy.testing.assert_array_equal(
+        modin_result, numpy_result, err_msg=f"Unary operator {operator} failed."
+    )
+
+
+def test_all_any_where():
```

Comment:
Could we add a test for `all` + `any` along an axis with a `where` argument?

Reply:
added!